src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/MeanQualityByCycleSparkIntegrationTest.java

```diff
@@ -21,6 +25,46 @@ public String getTestedClassName() {
         return MeanQualityByCycleSpark.class.getSimpleName();
     }
 
+    @DataProvider(name="filenames")
+    public Object[][] filenames() {
+        return new String[][]{
+                {"first5000a.bam", null},
+                {"first5000a.cram", b37_reference_20_21}
+        };
+    }
+    @Test(dataProvider="filenames", groups = {"R"})
+    public void test(final String inputFile, final String referenceName) throws IOException {
+        final File input = new File(TEST_DATA_DIR, inputFile);
+        final File expectedFile = new File(TEST_DATA_DIR, "meanqualbycycle.txt");
+        final File outfile = BaseTest.createTempFile("testMeanQualityByCycle", ".metrics");
+        final File pdf = BaseTest.createTempFile("testMeanQualityByCycle", ".pdf");
+        outfile.deleteOnExit();
+        pdf.deleteOnExit();
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.add("--input");
+        args.add(input.getAbsolutePath());
+        args.add("--output");
+        args.add(outfile.getAbsolutePath());
+        args.add("--chart");
+        args.add(pdf.getAbsolutePath());
+        if (null != referenceName) {
+            final File REF = new File(referenceName);
+            args.add("-R");
+            args.add(REF.getAbsolutePath());
+        }
+
+        runCommandLine(args.getArgsArray());
+
+        try (final FileReader actualReader = new FileReader(outfile);) {
+            final MetricsFile<?,Integer> output = new MetricsFile<>();
+            output.read(actualReader);
+            Assert.assertEquals(output.getAllHistograms().size(), 1);
+            Assert.assertEquals(output.getHistogram().size(), 202);
+        }
+        Assert.assertTrue(pdf.exists());
```

Comment:
added a check for pdf.length() > 0 everywhere where a pdf is expected (one place expects an empty pdf (when no output is created) - which is strange semantics but that's what it does)